### ISSUE TITLE
 Rookies not promoting after Gatecrasher, leading to crash in Avenger

### ISSUE THREAD

**Issue by ionbomb** [long-war-2/lwotc#92]
Playing alpha 2. 
Mods: Yet another F1, Evac All, Free Camera Rotation

Save file: [save_A2 Crash.zip](https://github.com/long-war-2/lwotc/files/3077970/save_A2.Crash.zip)

Steps:
Kill the final unit with the Reaper. The mission will end.
Click through to the squad debrief screen. Notice that no soldiers receive a promotion. All of them, except the Reaper, should have done so.
Go to the Barracks, and click "View Soldiers".
Game crashes.

**Comment by cicero225** [comment]
I didn't start an issue for this because I unfortunately didn't keep the save file foolishly, but I'd like to note that I also ended gatecrasher with an unusually low number of promotions (I had exactly 1).

No mods, started with a lost sitrep gatecrasher (which I thought might be related).

**Comment by ansabalirai** [comment]
@ionbomb From your savegame, I am able to get all the promotions. 
![20190414192056_1](https://user-images.githubusercontent.com/15093261/56104032-abd91680-5eea-11e9-915f-7a10bf4e61c3.jpg)
![20190414192100_1](https://user-images.githubusercontent.com/15093261/56104034-ae3b7080-5eea-11e9-8166-abd82e1d9b96.jpg)

Only thing I can think of is maybe disabling anarchy children and resistance armor pack dlc, since it gave me a warning when loading your savegame. Not sure what else could be causing this

**Comment by cicero225** [comment]
So I did another gatecrasher just to see this again, and yeah I can get the same result (though no crash). 
[save_NoPromotion.zip](https://github.com/long-war-2/lwotc/files/3078281/save_NoPromotion.zip)

Same thing, kill off the remaining sectoid and trooper, and get no promotions (I tested this by loading and doing it again).

It's possible this has something to do with building the mods wrong, as I also have the blacked out icons problem in #93

**Comment by pledbrook** [comment]
They're both working for me. The blacked out ability icons suggest a problem with the highlander. I'll reiterate what has been said elsewhere:

 1. Remove the LongWarOfTheChosen mod directly completely
 2. Make sure you have the latest alpha-2 which unpacks 2 mods: _LongWarOfTheChosen_ and _X2WOTCCommunityHighlander_ (unpack them to the same mods directory as before)
 3. Clear out your personal Config directory (you may have to start the game twice before Long War 2 is detected when you do this)
 4. Move you `SaveData/profile.bin` file to a safe backup location (you'll find it with the save files)
 5. Start the game!

XCOM 2 appears to have all sorts of weirdnesses that mean things break if directories and files aren't absolutely pristine. I suspect that's what's happening here and with the blacked out ability icons.

Hopefully once this is working with the two mods, future updates won't have these glitches.

**Comment by ionbomb** [comment]
So I worked out what the problem was.

I had followed the instructions telling me to remove any LongWarOfTheChosen or X2WOTCCommunityHighlander directory from the Mods folder (Steam\steamapps\common\XCOM 2\XCom2-WarOfTheChosen\XComGame\Mods) before unpacking the distribution.

However I left all my mods in the workshop folder (C:\...\Steam\steamapps\workshop\content\268500\) untouched. This included my prexisting steam version of the X2WOTCCommunityHighlander.

When I was running the game using the Vanilla launcher, with LongWarOfTheChosen and your modified X2WOTCCommunityHighlander ticked (as well as the three other QOL mods I have), it appears that the steam Highlander was being substituted for your modified version.

I realised this because whenever I went to relaunch the game after a crash or manual exit, the launcher had unticked your Highlander and ticked the steam version, as if those were the pre-selected settings.

I then tried using the AML, but it wouldn't let me select your Highlander without also selecting the steam version.

So I simply unsubscribed from the steam version of the Highlander, and now both issues has been resolved.
